src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx

```diff
@@ -0,0 +1,372 @@
+    themeSecondary: '#2b88d8',
+    themeDarkAlt: '#106ebe',
+    themeDark: '#005a9e',
+    themeDarker: '#004578',
+    neutralLighterAlt: '#f1f1f1',
+    neutralLighter: '#ededed',
+    neutralLight: '#e3e3e3',
+    neutralQuaternaryAlt: '#d3d3d3',
+    neutralQuaternary: '#cacaca',
+    neutralTertiaryAlt: '#c2c2c2',
+    neutralTertiary: '#c2c2c2',
+    neutralSecondary: '#858585',
+    neutralPrimaryAlt: '#4b4b4b',
+    neutralPrimary: '#333333',
+    neutralDark: '#272727',
+    black: '#1d1d1d',
+    white: '#f8f8f8',
+  },
+});
+
+const interval = 10000;
+
+export default class TaskRoleContainerList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      monacoProps: null,
+      monacoModalTitle: '',
+      logUrl: null,
+      type: '',
+    };
+
+    this.showSSHInfo = this.showSSHInfo.bind(this);
+    this.onDismiss = this.onDismiss.bind(this);
+    this.showContainerLog = this.showContainerLog.bind(this);
+    this.onRenderRow = this.onRenderRow.bind(this);
+    this.logAutoRefresh = this.logAutoRefresh.bind(this);
+    this.timer = null;
+  }
+
+  componentDidMount() {
+    this.startTimer();
+  }
+
+  componentWillUnmount() {
+    this.stopTimer();
+  }
+
+  logAutoRefresh() {
+    const {logUrl, type} = this.state;
+    if (logUrl === null) {
+      this.stopTimer();
+    } else {
+      void getContainerLog(logUrl, type).then((res) => {
+        const {logUrl: currentLogUrl} = this.state;
+        if (logUrl === currentLogUrl) {
+          this.setState({monacoProps: {value: res}});
+        }
+      }).catch((err) => {
+        const {logUrl: currentLogUrl} = this.state;
+        if (logUrl === currentLogUrl) {
+          this.setState({monacoProps: {value: err.message}});
+        }
+      });
+    }
+  }
+
+  startTimer() {
+    this.stopTimer();
+    this.timer = setInterval(this.logAutoRefresh, interval);
+  }
+
+  stopTimer() {
+    if (this.timer !== null) {
+      clearInterval(this.timer);
+      this.timer = null;
+    }
+  }
+
+  onDismiss() {
+    this.stopTimer();
+    this.setState({
+      monacoProps: null,
+      monacoModalTitle: '',
+      logUrl: null,
+      type: '',
+    });
+  }
+
+  showContainerLog(logUrl, type, title) {
+    this.setState({
+      monacoProps: {value: 'Loading...'},
+      monacoModalTitle: (
+        <div className={c(t.flex, t.justifyBetween)}>
+          <div>
+            {title}
+          </div>
+          <a
+            href={`${logUrl}${type}`}
+            target='_blank'
+            rel='noopener noreferrer'
+            className={c(localCss.fullLogLink, t.mr3)}
+          >
+            View Full Log
+          </a>
+        </div>
+      ),
+      logUrl,
+      type,
+    }, () => {
+      this.logAutoRefresh(); // start immediately
+      this.startTimer();
+    });
+  }
+
+  showSSHInfo(id) {
+    const {sshInfo} = this.props;
+    const containerSshInfo = sshInfo && sshInfo.containers.find((x) => x.id === id);
```

Comment:
I suggest to unify variable names with "ssh" in the middle.
Use `showSSHInfo` and `containerSSHInfo` when "ssh" is not at the start of the variable name, instead of `containerSshInfo` or `fetchSshInfo`.

Reply:
Previous job detail page (https://github.com/Microsoft/pai/blob/d26f762301f0de2efd263b84548f958b453d979b/src/webportal/src/app/job/job-view/job-view.component.js#L492) and rest server (https://github.com/Microsoft/pai/blob/d26f762301f0de2efd263b84548f958b453d979b/src/rest-server/src/controllers/job.js#L166) uses "Ssh". So I will unify function names to "Ssh"